src/OpenColorIO/Config.cpp

```diff
@@ -1078,6 +1094,435 @@ class Config::Impl
         // That should never happen.
         return -1;
     }
+
+    std::string getRefSpace(ConstConfigRcPtr & cfg) const
+    {
+        // Find a color space where isData is false and it has neither a to_ref or from_ref 
+        // transform.
+        auto nbCs = cfg->getNumColorSpaces();
+        for (int i = 0; i < nbCs; i++)
+        {
+            auto cs = cfg->getColorSpace(cfg->getColorSpaceNameByIndex(i));
+            if (cs->isData())
+            {
+                continue;
+            }
+
+            auto t = cs->getTransform(COLORSPACE_DIR_TO_REFERENCE);
+            if (t != nullptr)
+            {
+                continue;
+            }
+
+            t = cs->getTransform(COLORSPACE_DIR_FROM_REFERENCE);
+            if (t != nullptr) 
+            {
+                continue;
+            }
+
+            return cs->getName();
+        }
+        return "";
+    }
+
+    bool containsSRGB(ConstColorSpaceRcPtr & cs) const
+    {
+        std::string name = StringUtils::Lower(cs->getName());
+        if (StringUtils::Find(name, "srgb") != std::string::npos)
+        {
+            return true;
+        }
+
+        size_t nbOfAliases = cs->getNumAliases();
+        for (size_t i = 0; i < nbOfAliases; i++)
+        {
+            if (StringUtils::Find(cs->getAlias(i), "srgb") != std::string::npos)
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    ConstProcessorRcPtr getRefToSRGBTransform(ConstConfigRcPtr & builtinConfig, 
+                                              std::string refColorSpaceName) const
+    {
+        // Build reference space of the given prims to sRGB transform.
+        std::string srgbColorSpaceName = "Input - Generic - sRGB - Texture";
+
+        ColorSpaceTransformRcPtr csTransform = ColorSpaceTransform::Create();
+        csTransform->setSrc(refColorSpaceName.c_str());
+        csTransform->setDst(srgbColorSpaceName.c_str());
+        
+        ConstProcessorRcPtr proc  = getProcessorWithoutCaching(*builtinConfig,
+                                                               csTransform,
+                                                               TRANSFORM_DIR_FORWARD);
+        return proc;
+    }
+
+    bool isIdentityTransform(ProcessorRcPtr & proc, std::vector<float> & vals) const
+    {
+        std::vector<float> out = vals;
+
+        PackedImageDesc desc(&vals[0], (long) vals.size()/3, 1, CHANNEL_ORDERING_RGB);
+        PackedImageDesc descDst(&out[0], (long) vals.size()/3, 1, CHANNEL_ORDERING_RGB);
+
+        ConstCPUProcessorRcPtr cpu  = proc->getOptimizedCPUProcessor(OPTIMIZATION_LOSSLESS);
+        cpu->apply(desc, descDst);
+
+        for (size_t i = 0; i < out.size(); i++)
+        {
+            if (!EqualWithAbsError(vals[i], out[i], 1e-3f))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    std::string checkForLinearColorSpace(ConstColorSpaceRcPtr & cs,
```

Comment:
Suggest renaming to make it more clear what the function is doing, `getReferenceSpaceFromLinearSpace` and similarly `getReferenceSpaceFromSRGBSpace`?

Reply:
Just for clarity, the second proposal was to update the corresponding `checkForSRGBTextureColorSpace` name.

Reply:
Ah, sorry. I read that wrong. I'll change the other one also as it makes more sense.